File: app/src/main/java/com/android/benchmark/ui/ShadowGridActivity.java
/*
 * Copyright (C) 2015 The Android Open Source Project
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *      http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */
package com.android.benchmark.ui;

import android.content.Intent;
import android.os.Bundle;
import androidx.fragment.app.FragmentManager;
import androidx.fragment.app.ListFragment;
import androidx.appcompat.app.AppCompatActivity;
import android.view.View;
import android.widget.ArrayAdapter;
import android.widget.ListView;

import com.android.benchmark.R;
import com.android.benchmark.ui.automation.Automator;
import com.android.benchmark.ui.automation.Interaction;

public class ShadowGridActivity extends AppCompatActivity {
    private Automator mAutomator;
    public static class MyListFragment extends ListFragment {
	    @Override
	    public void onViewCreated(final View view, final Bundle savedInstanceState) {
		    super.onViewCreated(view, savedInstanceState);
            this.getListView().setDivider(null);
	    }
    }

    @Override
    protected void onCreate(final Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        int runId = this.getIntent().getIntExtra("com.android.benchmark.RUN_ID", 0);
        int iteration = this.getIntent().getIntExtra("com.android.benchmark.ITERATION", -1);

        final FragmentManager fm = this.getSupportFragmentManager();
        if (null == fm.findFragmentById(android.R.id.content)) {
            final ListFragment listFragment = new MyListFragment();

            listFragment.setListAdapter(new ArrayAdapter<>(this,
                    R.layout.card_row, R.id.card_text, Utils.buildStringList(200)));
            fm.beginTransaction().add(android.R.id.content, listFragment).commit();

            final String testName = this.getString(R.string.shadow_grid_name);

            this.mAutomator = new Automator(testName, runId, iteration, this.getWindow(),
                    new Automator.AutomateCallback() {
                @Override
                public void onPostAutomate() {
                    final Intent result = new Intent();
                    ShadowGridActivity.this.setResult(Activity.RESULT_OK, result);
                    ShadowGridActivity.this.finish();
                }

                @Override
                public void onAutomate() {
                    final ListView v = ShadowGridActivity.this.findViewById(android.R.id.list);

                    final int[] coordinates = new int[2];
                    v.getLocationOnScreen(coordinates);

                    final int x = coordinates[0];
                    final int y = coordinates[1];

                    final float width = v.getWidth();
                    final float height = v.getHeight();

                    final float middleX = (x + width) / 2;
                    final float middleY = (y + height) / 2;

                    final Interaction flingUp = Interaction.newFlingUp(middleX, middleY);
                    final Interaction flingDown = Interaction.newFlingDown(middleX, middleY);

                    this.addInteraction(flingUp);
                    this.addInteraction(flingDown);
                    this.addInteraction(flingUp);
                    this.addInteraction(flingDown);
                    this.addInteraction(flingUp);
                    this.addInteraction(flingDown);
                    this.addInteraction(flingUp);
                    this.addInteraction(flingDown);
                }
            });
            this.mAutomator.start();
        }
    }

    @Override
    protected void onPause() {
        super.onPause();
        if (null != mAutomator) {
            this.mAutomator.cancel();
            this.mAutomator = null;
        }
    }
}

File: app/src/main/java/com/android/benchmark/ui/ShadowGridActivity.kt
/*
 * Copyright (C) 2015 The Android Open Source Project
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *      http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */
package com.android.benchmark.ui

import android.content.Intentimport

android.os.Bundleimport android.view.Viewimport android.widget.ArrayAdapterimport android.widget.ListViewimport androidx.appcompat.app.AppCompatActivityimport androidx.fragment.app.ListFragmentimport com.android.benchmark.Rimport com.android.benchmark.ui.automation.Automatorimport com.android.benchmark.ui.automation.Automator.AutomateCallbackimport com.android.benchmark.ui.automation.Interaction
class ShadowGridActivity : AppCompatActivity() {
    private var mAutomator: Automator? = null

    class MyListFragment : ListFragment() {
        override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
            super.onViewCreated(view, savedInstanceState)
            listView.divider = null
        }
    }

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        val runId = intent.getIntExtra("com.android.benchmark.RUN_ID", 0)
        val iteration = intent.getIntExtra("com.android.benchmark.ITERATION", -1)
        val fm = supportFragmentManager
        if (null == fm.findFragmentById(android.R.id.content)) {
            val listFragment: ListFragment = MyListFragment()
            listFragment.listAdapter = ArrayAdapter<String?>(this,
                    R.layout.card_row, R.id.card_text, Utils.Companion.buildStringList(200))
            fm.beginTransaction().add(android.R.id.content, listFragment).commit()
            val testName = getString(R.string.shadow_grid_name)
            mAutomator = Automator(testName, runId, iteration, window,
                    object : AutomateCallback() {
                        override fun onPostAutomate() {
                            val result = Intent()
                            setResult(RESULT_OK, result)
                            finish()
                        }

                        override fun onAutomate() {
                            val v = findViewById<ListView>(android.R.id.list)
                            val coordinates = IntArray(2)
                            v.getLocationOnScreen(coordinates)
                            val x = coordinates[0]
                            val y = coordinates[1]
                            val width = v.width.toFloat()
                            val height = v.height.toFloat()
                            val middleX = (x + width) / 2
                            val middleY = (y + height) / 2
                            val flingUp: Interaction = Interaction.Companion.newFlingUp(middleX, middleY)
                            val flingDown: Interaction = Interaction.Companion.newFlingDown(middleX, middleY)
                            addInteraction(flingUp)
                            addInteraction(flingDown)
                            addInteraction(flingUp)
                            addInteraction(flingDown)
                            addInteraction(flingUp)
                            addInteraction(flingDown)
                            addInteraction(flingUp)
                            addInteraction(flingDown)
                        }
                    })
            mAutomator!!.start()
        }
    }

    override fun onPause() {
        super.onPause()
        if (null != mAutomator) {
            mAutomator!!.cancel()
            mAutomator = null
        }
    }
}